### PR TITLE
[sBTC DR] filter by sBTC wallet address

### LINKED ISSUE #167: [sBTC DR] verify sbtc wallet address in btc txs
Only handle btc deposit and withdrawal txs for the current sbtc wallet address.

### PR DESCRIPTION
**Pull request by friedger**
## Description

This PR
* filters deposit and withdrawal requests by sbtc wallet address
* fixes #167

### CODE REVIEW

File: romeo/src/state.rs [hunk 1]
```diff
@@ -232,10 +238,23 @@ fn process_bitcoin_block(
...
+			.collect::<Vec<_>>(),
+	);
+	state.withdrawals.extend(
+		parse_withdrawals(config, &block)
+			.into_iter()
+			.filter(|withdrawal| {
+				withdrawal.info.sbtc_wallet_address == sbtc_wallet_address
+			})
+			.collect::<Vec<_>>(),
+	);
```
Comment (CAGS295) [inline romeo/src/state.rs:257]: Filter implements [IntoIterator](https://doc.rust-lang.org/std/iter/struct.Filter.html#impl-IntoIterator-for-Filter%3CI,+P%3E). No need to collect into a Vec<_>.
```suggestion
		state.deposits.extend(
		parse_deposits(config, bitcoin_height, &block)
			.into_iter()
			.filter(|deposit| {
				deposit.info.sbtc_wallet_address == sbtc_wallet_address
			}),
	);
	state.withdrawals.extend(
		parse_withdrawals(config, &block)
			.into_iter()
			.filter(|withdrawal| {
				withdrawal.info.sbtc_wallet_address == sbtc_wallet_address
			}),
	);
```

**Review by CAGS295 — CHANGES_REQUESTED** [review]

File: romeo/src/state.rs (hunk 1 above)
Comment (friedger) [reply romeo/src/state.rs]: adopted

File: romeo/src/state.rs [hunk 2]
```diff
@@ -245,15 +245,15 @@ fn process_bitcoin_block(
 			.filter(|deposit| {
 				deposit.info.sbtc_wallet_address == sbtc_wallet_address
 			})
-			.collect::<Vec<_>>(),
+			.into_iter(),
```
Comment (CAGS295) [inline romeo/src/state.rs]:
```suggestion
```
Does this work?

**Review by CAGS295 — DISMISSED** [review]

**Review by stjepangolemac — COMMENTED** [review]
Given that `parse_deposits()` and `parse_withdrawals()` already accepts config I'd prefer the filter to live there. That allows us to not add any new fields to the existing structures.

**Comment by friedger** [comment]
@stjepangolemac Moved filtering to parse functions in 7deebc1b3605587910470965a45f48b5e3324fea

**Review by stjepangolemac — APPROVED** [review]

**Review by CAGS295 — DISMISSED** [review]

**Review by CAGS295 — APPROVED** [review]

**Review by jcnelson — APPROVED** [review]

**Pull request merged**